shell/platform/embedder/embedder_surface_vulkan.cc

```diff
@@ -0,0 +1,141 @@
+      !vulkan_dispatch_table_.get_next_image ||
+      !vulkan_dispatch_table_.present_image) {
+    return;
+  }
+
+  vk_->SetupInstanceProcAddresses(vulkan::VulkanHandle<VkInstance>{instance});
+  vk_->SetupDeviceProcAddresses(vulkan::VulkanHandle<VkDevice>{device});
+  if (!vk_->IsValid()) {
+    FML_LOG(ERROR) << "VulkanProcTable invalid.";
+    return;
+  }
+
+  main_context_ = CreateGrContext(instance, ContextType::kRender);
+  // TODO(bdero): Add a second (optional) queue+family index to the Embedder API
+  //             to allow embedders to specify a dedicated transfer queue for
+  //             use by the resource context. Queue families with graphics
+  //             capability can always be used for memory transferring, but it
+  //             would be adventageous to use a dedicated transter queue here.
+  resource_context_ = CreateGrContext(instance, ContextType::kResource);
+
+  valid_ = main_context_ && resource_context_;
+}
+
+EmbedderSurfaceVulkan::~EmbedderSurfaceVulkan() = default;
+
+// |GPUSurfaceVulkanDelegate|
+const vulkan::VulkanProcTable& EmbedderSurfaceVulkan::vk() {
+  return *vk_;
+}
+
+// |GPUSurfaceVulkanDelegate|
+VkImage EmbedderSurfaceVulkan::AcquireImage(const SkISize& size) {
+  return vulkan_dispatch_table_.get_next_image(size);
+}
+
+// |GPUSurfaceVulkanDelegate|
+bool EmbedderSurfaceVulkan::PresentImage(VkImage image) {
+  return vulkan_dispatch_table_.present_image(image);
+}
+
+// |EmbedderSurface|
+bool EmbedderSurfaceVulkan::IsValid() const {
+  return valid_;
+}
+
+// |EmbedderSurface|
+std::unique_ptr<Surface> EmbedderSurfaceVulkan::CreateGPUSurface() {
+  const bool render_to_surface = !external_view_embedder_;
+  return std::make_unique<GPUSurfaceVulkan>(this, main_context_,
+                                            render_to_surface);
+}
+
+// |EmbedderSurface|
+sk_sp<GrDirectContext> EmbedderSurfaceVulkan::CreateResourceContext() const {
+  return resource_context_;
+}
+
+sk_sp<GrDirectContext> EmbedderSurfaceVulkan::CreateGrContext(
+    VkInstance instance,
+    ContextType context_type) const {
+  uint32_t skia_features = 0;
+  if (!device_.GetPhysicalDeviceFeaturesSkia(&skia_features)) {
+    FML_LOG(ERROR) << "Failed to get physical device features.";
+
+    return nullptr;
+  }
+
+  auto get_proc = vk_->CreateSkiaGetProc();
+  if (get_proc == nullptr) {
+    FML_LOG(ERROR) << "Failed to create Vulkan getProc for Skia.";
+    return nullptr;
+  }
+
+  GrVkExtensions extensions;
+
+  GrVkBackendContext backend_context = {};
+  backend_context.fInstance = instance;
+  backend_context.fPhysicalDevice = device_.GetPhysicalDeviceHandle();
+  backend_context.fDevice = device_.GetHandle();
+  backend_context.fQueue = device_.GetQueueHandle();
+  backend_context.fGraphicsQueueIndex = device_.GetGraphicsQueueIndex();
+  backend_context.fMinAPIVersion = VK_MAKE_VERSION(1, 0, 0);
+  backend_context.fMaxAPIVersion = VK_MAKE_VERSION(1, 0, 0);
```

Comment:
Fuchsia requires at least VK_MAKE_VERSION(1, 1, 0) here -- can we make this configurable via FlutterVulkanRendererConfig?

Reply:
I noticed that memory_requirements_2 is a core extension in 1.1, so I changed it to activate automatically when the version is >= 1.1.0. Let me know if this seems good. :)